docs/_writing_macros.rst

```diff
@@ -248,32 +248,94 @@ The following expand declarations have replaced the original conditional element
           </options>
         </expand>
 
+From Galaxy release 22.01 named yields are supported. That is, if the macro contains
+``<yield name="xyz"/>`` it is replaced by the content of the ``token` tag with the same name.
+Token tags need to be direct children of the ``expand`` element.
+
+.. code-block:: xml
+
+    <xml name="named_yields_example">
+      <conditional>
+        <param type="select">
+          <option value="a">A</option>
+          <option value="b">B</option>
+          <yield name="more_options"/>
+        </param>
+        <when value="a">
+          <param type="select">
+            <yield />
+          </param>
+        </when>
+        <when value="b">
+          <param ... />
+        </when>
+        <yield name="more_whens">
+      </conditional>
+    </xml>
+
+.. code-block:: xml
+
+        <expand macro="named_yields_example">
+          <token name="more_options">
+            <option value="c">C</option>
+          </token>
+          <token name="more_whens">
+            <when value="c">
+              <param type="select">
+                <yield />
+              </param>
+            </when>
+          </token>
+          <options from_data_table="tophat2_indexes">
+            <filter type="sort_by" column="2"/>
+            <validator type="no_options" message="No genomes are available for the selected input dataset"/>
+          </options>
```

Comment:
```suggestion
```
This is unrelated to the macro example, right ?

Reply:
The unnamed yield is replaced by this.

Reply:
Oh, I have not understood this. Why is that ? How do I read this as a tool author ?

Reply:
Essentially as before: The state was that the content of the expand tag is inserted at the place(s) of the yield tags in the macro. 

In order to keep backward compatibility I implemented it such still all content of the expand minus the token tags and their content is used for the replacement.

Reply:
Ah, I see, yeah, that makes sense.